### PR TITLE
Hide controlling BMS setting if BMS is unavailable

### LINKED ISSUE #1078: Hide Controlling BMS setting if no BMS
gui-v1 commit: 85453f72d8365297aaecbb845a084f7e2a65c63f

### PR DESCRIPTION
**Pull request by jpetrell**
Fixes #1078.

### CODE REVIEW

File: pages/settings/PageSettingsDvcc.qml [hunk 1]
```diff
@@ -155,7 +155,8 @@ Page {
 				//: Shown when BMS instance is invalid
 				//% "Unavailable, set another"
 				defaultSecondaryText: qsTrId("settings_dvcc_unavailable_bms")
-
+				allowed: defaultAllowed && commonSettings.dvccActive
+						 && (bmsOptions.defaultOptionModel.length > 2 || bmsService.isValid)
```
Comment (blammit) [inline pages/settings/PageSettingsDvcc.qml:159]: The `defaultOptionModel` is read-only and always has a length of 2. This should check `bmsOptions.optionModel.length` instead.

**Review by blammit — CHANGES_REQUESTED** [review]

File: pages/settings/PageSettingsDvcc.qml (hunk 1 above)
Comment (jpetrell) [reply pages/settings/PageSettingsDvcc.qml]: Good catch! :coin:  Fixed.

**Review by blammit — APPROVED** [review]

File: pages/settings/PageSettingsDvcc.qml [hunk 2]
```diff
@@ -155,7 +155,8 @@ Page {
 				//: Shown when BMS instance is invalid
 				//% "Unavailable, set another"
 				defaultSecondaryText: qsTrId("settings_dvcc_unavailable_bms")
-
+				allowed: defaultAllowed && commonSettings.dvccActive
+						 && (bmsOptions.optionModel.length > 2 || bmsService.isValid)
```
Comment (chriadam) [inline pages/settings/PageSettingsDvcc.qml:159]: this > 2 check seems a bit magic to me.  Could a comment be added?

Comment (jpetrell) [reply pages/settings/PageSettingsDvcc.qml:160]: Added.

**Pull request merged**